### PR TITLE
ImportC: __attribute((deprecated)) and __declspec(deprecated)

### PR DESCRIPTION
**Pull request by WalterBright**
As suggested by @ibuclaw https://issues.dlang.org/show_bug.cgi?id=21938

It just converts it to the D deprecate.

### CODE REVIEW

**Comment by WalterBright** [comment]
This took more work than I expected. Anyhow, it's good to go.

File: compiler/test/compilable/stdcheaders.c [hunk 1]
```diff
@@ -80,6 +80,9 @@
 #endif
 #endif
 
+#ifdef _MSC_VER
+#define _CRT_NONSTDC_NO_DEPRECATE 1
+#endif
```
Comment (PetarKirov) [inline compiler/test/compilable/stdcheaders.c:85]: Is this still necessary after this PR?

File: druntime/src/importc.h [hunk 2]
```diff
@@ -109,6 +109,7 @@
 #if _MSC_VER
 //#undef _Post_writable_size
 //#define _Post_writable_size(x) // consider #include <no_sal2.h>
+#define _CRT_INSECURE_DEPRECATE(x)
```
Comment (PetarKirov) [inline druntime/src/importc.h:112]: Why is this necessary? Shouldn't it be up to the user to decide whether to disable this warning?

**Review by PetarKirov — APPROVED** [review]
Apart from the question about those two defines, the implementation looks good to me!

File: druntime/src/importc.h (hunk 2 above)
Comment (WalterBright) [reply druntime/src/importc.h:112]: Not doing this turns on a lot of inline deprecated functions, and a blizzard of deprecation messages come out.

File: compiler/test/compilable/stdcheaders.c (hunk 1 above)
Comment (WalterBright) [reply compiler/test/compilable/stdcheaders.c]: no, removed

**Comment by PetarKirov** [comment]
@WalterBright feel free to merge manually.
